File: base/Checking.java
import java.util.*;

public class Checking{
	Scanner input = new Scanner(System.in);
	protected double balance;
	protected double amount;

	public static void main(String[] args){
		new Checking();	
	}// end main


	public Checking(){
		this.amount = 0;
	}// end constructor

	public void setBalance(double balance){
		this.balance = balance;
	} // end setter

	public void deposit(){
		System.out.print("Please enter the amount you want to deposit: ");
		this.amount = input.nextDouble();
		this.balance += this.amount;
		System.out.println("Success! Your balance is now " + this.balance + ".");
	}// end deposit

	public void withdraw(){
		System.out.print("Please enter the amount you want to withdraw: ");
		this.amount = input.nextDouble();
		if(this.amount > this.balance){
			System.out.println("You only have " + this.balance + " in you checking account. You cannot withdraw more than that.");
		}
		else{
			this.balance -= this.amount;
			System.out.println("Success! Your balance is now " + this.balance + ".");
	}
	}// end withdraw

	public double getBalance(){
		return this.balance;
	} // end getter
}

File: blackbelt/checking.cpp
#include "checking.h"
#include <iostream>
#include <string>

Checking::Checking(){

	// Setting default values
	Checking::balance = 100.0;
	Checking::amount = 0.0;
}

void Checking::setBalance(double balance){
	Checking::balance = balance;
}

void Checking::deposit(){
	std::cout<<"Please enter the amount you want to deposit: ";
	std::cin>>Checking::amount;
	Checking::balance += Checking::amount;
	std::cout<<"Success! Your balance is now " << Checking::balance << std::endl;
}

void Checking::withdraw(){
	std::cout<<"Please enter the amount you want to withdraw: ";
	std::cin>>Checking::amount;
	if(Checking::amount > Checking::balance){
		std::cout<<"You only have " << Checking::balance << " in your checking account. You cannot withdraw more than that."<<std::endl;
	}
	else{
		Checking::balance -= Checking::amount;
		std::cout<<"Success! Your balance is now " << Checking::balance <<std::endl;
	}
}

double Checking::getBalance(){
	return Checking::balance;
}


